File: Assignment8 java/src/Account.java
public class Account {
    private double balance;
    private String accountNumber;

    public Account(String accNumber, double initialBalance) {
        accountNumber = accNumber;
        balance = initialBalance;
    }

    public void credit(double amount) {
        balance += amount;
    }

    public void debit(double amount) {
        if (amount > balance) {
            System.out.println("Debit amount exceeded account balance.");
        } else {
            balance -= amount;
        }
    }

    public double getBalance() {
        return balance;
    }

    public String getAccountNumber() {
        return accountNumber;
    }

    public void setAccountNumber(String accNumber) {
        accountNumber = accNumber;
    }

    public void displayAccountInfo() {
        System.out.println("Account number: " + accountNumber);
        System.out.println("Balance: " + balance);
    }
}

File: Account.cpp
#include<iostream>;
#include<string>;
#include"Account.h";
int main() {
    Account account1("122100008121", 1000050);
    Account account2("122100008122", 2000050);

    cout << "Initial Account Information:\n";
    cout << "Account Number: " << account1.getAccountNumber() << ", Balance: $" << account1.getBalance() << endl;
    cout << "Account Number: " << account2.getAccountNumber() << ", Balance: $" << account2.getBalance() << endl;

    cout << "\nAccount 1 Transactions:\n";
    account1.credit(1000000);
    cout << "After credit, Balance: $" << account1.getBalance() << endl;
    account1.debit(2000000);
    cout << "After debit, Balance: $" << account1.getBalance() << endl;

    cout << "\nAccount 2 Transactions:\n";
    account2.credit(1000000);
    cout << "After credit, Balance: $" << account2.getBalance() << endl;
    account2.debit(2000000);
    cout << "After debit, Balance: $" << account2.getBalance() << endl;

    return 0;
}